docs/2.7.0/docs/canton/usermanual/security.rst

```diff
@@ -550,6 +550,24 @@ AWS KMS only supports the :ref:`following cryptographic schemes <canton_supporte
     provider it can no longer be reverted without a full reset of the node
     (i.e., re-generation of node identity and all keys).
 
+In the previous example, Canton creates its own keys on startup. If we want to create the keys ourselves and pass then
+to Canton we have to manually start the nodes by adding the following flag in the config:
+
+.. code-block:: none
+
+    <node>.init.auto-init = false
+
+We then need to register the keys in Canton by running the key registration command on each node.
+For example for a participant we would run:
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/app/src/test/resources/kms-provider-tagged.conf
+   :language: scala
+   :start-after: user-manual-entry-begin: ManualRegisterKmsKeys
+   :end-before: user-manual-entry-end: ManualRegisterKmsKeys
+
+Finally, we need to initialize our :ref:`domain <_manually-init-domain>` and
```

Comment:
those steps still include the generation of keys. can you refactor them such that either the three keys are generated or registered at the beginning, and then all follow-up steps are independent of the key setup step. and include the common steps also here

Reply:
ah right, my bad, I was looking at the v2.6.5 docs